### PR TITLE
Add Azure Provider Support

### PR DESCRIPTION
**Pull request by felixrieseberg**
Hey everyone, your friendly Microsoft Open Source team with a little PR - adding support for Azure. We understand that stuff is still in development, but getting Azure into the development pipeline earlier rather than later is probably a good idea.

 * Full compile/deploy/remove/invoke support
 * Unit tests where appropriate
 * No integration tests (yet)

### CODE REVIEW

**Comment by ac360** [comment]
Awesome work @felixrieseberg @bnookala @timfpark @mattpodwysocki !  Really well done guys :)

**Comment by pmuens** [comment]
This is great! Thank you so much guys! We'll look into it ASAP!

**Comment by eahefnawy** [comment]
Sweeeet! Thanks tons @felixrieseberg 😊 💃 👯

File: lib/plugins/azure/compile/index.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        throw new this.serverless.classes.Error(
+          `Function ${functionName} does not have an azure trigger configuration`
+        );
+      }
+
+      // Identify the type of trigger to deploy.
+      if (functionObject.events.azure.http_endpoint) {
+        functionJSON = HttpTrigger.buildFunctionJSON(functionObject);
+      } else if (functionObject.events.azure.scheduled_trigger) {
+        functionJSON = TimerTrigger.buildFunctionJSON(TimerTrigger);
```
Comment (jthomas) [inline lib/plugins/azure/compile/index.js]: I might be missing the context here but should this be....
```
functionJSON = TimerTrigger.buildFunctionJSON(functionObject);
```

It looks like you're passing in the TimerTrigger object to itself?

Comment (bnookala) [inline lib/plugins/azure/compile/index.js]: @jthomas D'oh. This is me. I'll fix this!

**Comment by pmuens** [comment]
Thanks for this amazing work! We've recently pulled it into a local branch and will update / work on it there (see https://github.com/serverless/serverless/pull/1547).

**Pull request closed without merging**